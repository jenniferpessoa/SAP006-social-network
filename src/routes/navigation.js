export function navigation(path) {
  window.history.pushState({}, '', path);
  const popStateEvent = new PopStateEvent('popstate', { state: {} });
  dispatchEvent(popStateEvent);
}
