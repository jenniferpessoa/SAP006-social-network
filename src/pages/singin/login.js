import { getError } from '../../Errors/index.js';
import { navigation } from '../../routes/navigation.js';
import { loginEmailAndPassword, loginWithGmail } from '../../services/index.js';

export const Login = () => {
  const root = document.createElement('div');
  root.innerHTML = `
  <header class='logo'>
  <img src="../.././img/logo-start.png"  height="80px" width="350px" alt="">
    <h2>A Rede Social dos Velejadores</h2>
  </header>
  <main class='container'>  
    <section class='img-container'>
    </section>    
    <section id='login' class='login-container'>
      <header class='form-options'>
        <span class='opt-login'>Login</span>
        <span class='opt-signup'>Cadastro</span>
      </header>
      <section class='form-container'>
        <form>
          <div class='form-fields'>
            <label for='email'>E-mail</label>
            <input id='email' type='e-mail' class='input-email form-item'>
              
            <label class='label-login' for='password'>Senha</label>
            <input id='password' type='password' class='input-password form-item'>
          </div>
          <section class='errors'></section>
          <div class='rememberForgot' >
            <a href='#' id='reset'>Esqueci a senha</a>
          </div>  

          <button type='button' id='buttonLogin' class='btn-login btn form-item'>Entrar</button>
          <p class='separator'>ou</p>                      
          <button type='button' id='btnGmail' class='btnGmail btn form-item'>
            <img src='./img/logo-google.png' class='google-icon'></img>
            <span>Entrar com o Google</span>
          </button>
        </form>
      </section>
    </section>
  </main>
  `;

  const reset = root.querySelector('#reset');
  const btnSignUp = root.querySelector('.opt-signup');
  const btnLogin = root.querySelector('#buttonLogin');
  const btnGmail = root.querySelector('#btnGmail');
  const inputEmail = root.querySelector('#email');
  const inputPassword = root.querySelector('#password');

  reset.addEventListener('click', (event) => {
    event.preventDefault();
    navigation('/reset');
  });

  btnSignUp.addEventListener('click', () => navigation('/signup'));

  btnLogin.addEventListener('click', () => {
    loginEmailAndPassword(inputEmail.value, inputPassword.value);
  });

  btnGmail.addEventListener('click', () => {
    loginWithGmail().then(() => {
      navigation('/feed');
    }).catch((error) => {
      getError(error);
    });
  });

  return root;
};
