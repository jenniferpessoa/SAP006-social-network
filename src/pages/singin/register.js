import { signUpWithEmailAndPassword, loginWithGmail } from '../../services/index.js';
import { navigation } from '../../routes/navigation.js';
import { getError } from '../../Errors/index.js';

export const SignUp = () => {
  const root = document.createElement('div');
  root.innerHTML = `
  <header class='logo'>
    <h1>A Bordo</h1>
    <h2>A Rede Social dos Velejadores</h2>
  </header>
  <main id='loginPage' class='container'>
    <section class='img-container'></section>    
    <section id='login' class='login-container'>
      <header class='form-options'>
        <span class='opt-login'>Login</span>
        <span class='opt-signup linkSignUp'>Cadastro</span>
      </header>
      <section class='form-container'>
        <form>
          <div class='form-fields'>
            <label for='email'>E-mail</label>
            <input id='email' type='e-mail' class='input-email form-item'>
            
            <label class='label-login' for='password'>Senha</label>
            <input id='password' type='password' class='input-password form-item'>
          </div>
          <section class='errors'></section>
          <button type='button' id='signUpButton' class='btn-signup btn form-item'>Cadastrar</button>
          <p class='separator'>ou</p> 
          <button type='button' id='btnGmail' class='btnGmail btn form-item'>
            <img src='./img/logo-google.png' class='google-icon'></img>
            <span>Entrar com o Google</span>
          </button>
        </form>
      </section>
    </section>
  </main>
  `;

  const btnLogin = root.querySelector('.opt-login');
  const btnSignUp = root.querySelector('#signUpButton');
  const btnGoogle = root.querySelector('#btnGmail');
  const inputEmail = root.querySelector('#email');
  const inputPassword = root.querySelector('#password');

  btnLogin.addEventListener('click', () => {
    navigation('/');
  });

  btnSignUp.addEventListener('click', () => {
    signUpWithEmailAndPassword(inputEmail.value, inputPassword.value)
      .then(() => {
        navigation('/profile');
      })
      .catch((error) => {
        getError(error);
      });
  });

  btnGoogle.addEventListener('click', () => {
    loginWithGmail();
  });

  return root;
};
