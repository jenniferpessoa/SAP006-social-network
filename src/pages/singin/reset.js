/* eslint-disable no-alert */
import { getError } from '../../Errors/index.js';
import { navigation } from '../../routes/navigation.js';
import { resetPassword } from '../../services/index.js';

export const Reset = () => {
  const root = document.createElement('div');
  root.innerHTML = `
  <header class='logo'>
    <h1>A Bordo</h1>
    <h2>A Rede Social dos Velejadores</h2>
  </header>
  <main class='container'>  
    <section class='img-container'>
    </section>    
    <section id='login' class='login-container'>
      <header class='reset-text'>
        Redefinir Senha
      </header>
      <section class='form-container'>
        <form>
          <div class='form-fields'>
            <label for='email'>E-mail</label>
            <input id='email' type='e-mail' class='input-item'>
              
            <button type='button' id='buttonReset' class='btn-login btn form-item'>Enviar</button>

            <button type='button' id='buttonReturn' class='btn-login btn form-item'>Voltar</button>
          </div>  
        </form>
      </section>
    </section>
  </main>
  `;

  const btnReset = root.querySelector('#buttonReset');
  const btnReturn = root.querySelector('#buttonReturn');
  const inputEmail = root.querySelector('#email');

  btnReset.addEventListener('click', () => {
    resetPassword(inputEmail.value).then(() => {
      alert('E-mail enviado com sucesso! Confira sua caixa de entrada');
    }).catch((error) => {
      getError(error);
    });
  });

  btnReturn.addEventListener('click', () => {
    navigation('/');
  });

  return root;
};
