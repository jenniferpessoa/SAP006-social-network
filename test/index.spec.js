/**
 * @jest-environment jsdom
 */

import { Login } from '../src/pages/singin/login.js';
import { SignUp } from '../src/pages/singin/register.js';
import { Reset } from '../src/pages/singin/reset.js';
import * as services from '../src/services/index.js';
import { navigation } from '../src/routes/navigation.js';

jest.mock('../src/services/index.js');
jest.mock('../src/routes/navigation.js');
window.alert = jest.fn();

describe('Login', () => {
  const loginPage = Login();
  it('should load', () => {
    expect(Login()).toMatchSnapshot();
  });
  it('should login with email and password', () => {
    services.loginEmailAndPassword.mockResolvedValueOnce(true);
    loginPage.querySelector('#email').value = 'teste@teste.com';
    loginPage.querySelector('#password').value = '12345678';
    loginPage.querySelector('#buttonLogin').dispatchEvent(new Event('click'));
    expect(services.loginEmailAndPassword).toBeCalled();
    expect(services.loginEmailAndPassword).toHaveBeenCalledWith('teste@teste.com', '12345678');
  });

  it('should login with google account', () => {
    services.loginWithGmail.mockResolvedValueOnce(true);
    loginPage.querySelector('#btnGmail').dispatchEvent(new Event('click'));
    expect(services.loginWithGmail).toBeCalled();
  });
  it('should redirect to reset password page', () => {
    loginPage.querySelector('#reset').dispatchEvent(new Event('click'));
    expect(navigation).toBeCalledWith('/reset');
  });
});

describe('register', () => {
  const signUpPage = SignUp();
  it('should load page', () => {
    expect(SignUp()).toMatchSnapshot();
  });
  it('should siogn up with email and password', () => {
    services.signUpWithEmailAndPassword.mockResolvedValueOnce(true);
    signUpPage.querySelector('#email').value = 'teste@teste.com';
    signUpPage.querySelector('#password').value = '12345678';
    signUpPage.querySelector('#signUpButton').dispatchEvent(new Event('click'));
    expect(services.signUpWithEmailAndPassword).toBeCalled();
    expect(services.signUpWithEmailAndPassword).toHaveBeenCalledWith('teste@teste.com', '12345678');
  });
  it('should redirect to login  page', () => {
    signUpPage.querySelector('.opt-login').dispatchEvent(new Event('click'));
    expect(navigation).toBeCalledWith('/');
  });
});

describe('reset', () => {
  const resetPage = Reset();
  it('should load reset page', () => {
    expect(Reset()).toMatchSnapshot();
  });
  it('should siogn up with email and password', () => {
    window.alert.mockClear();
    services.resetPassword.mockResolvedValueOnce(true);
    resetPage.querySelector('#email').value = 'teste@teste.com';
    resetPage.querySelector('#buttonReset').dispatchEvent(new Event('click'));
    expect(services.resetPassword).toBeCalled();
    expect(services.resetPassword).toHaveBeenCalledWith('teste@teste.com');
  });
});
