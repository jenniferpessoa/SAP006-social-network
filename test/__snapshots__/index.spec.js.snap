// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should load 1`] = `
<div>
  
  
  <header
    class="logo"
  >
    
  
    <img
      alt=""
      height="80px"
      src="../.././img/logo-start.png"
      width="350px"
    />
    
    
    <h2>
      A Rede Social dos Velejadores
    </h2>
    
  
  </header>
  
  
  <main
    class="container"
  >
      
    
    <section
      class="img-container"
    >
      
    
    </section>
        
    
    <section
      class="login-container"
      id="login"
    >
      
      
      <header
        class="form-options"
      >
        
        
        <span
          class="opt-login"
        >
          Login
        </span>
        
        
        <span
          class="opt-signup"
        >
          Cadastro
        </span>
        
      
      </header>
      
      
      <section
        class="form-container"
      >
        
        
        <form>
          
          
          <div
            class="form-fields"
          >
            
            
            <label
              for="email"
            >
              E-mail
            </label>
            
            
            <input
              class="input-email form-item"
              id="email"
              type="e-mail"
            />
            
              
            
            <label
              class="label-login"
              for="password"
            >
              Senha
            </label>
            
            
            <input
              class="input-password form-item"
              id="password"
              type="password"
            />
            
          
          </div>
          
          
          <section
            class="errors"
          />
          
          
          <div
            class="rememberForgot"
          >
            
            
            <a
              href="#"
              id="reset"
            >
              Esqueci a senha
            </a>
            
          
          </div>
            

          
          <button
            class="btn-login btn form-item"
            id="buttonLogin"
            type="button"
          >
            Entrar
          </button>
          
          
          <p
            class="separator"
          >
            ou
          </p>
                                
          
          <button
            class="btnGmail btn form-item"
            id="btnGmail"
            type="button"
          >
            
            
            <img
              class="google-icon"
              src="./img/logo-google.png"
            />
            
            
            <span>
              Entrar com o Google
            </span>
            
          
          </button>
          
        
        </form>
        
      
      </section>
      
    
    </section>
    
  
  </main>
  
  
</div>
`;

exports[`register should load page 1`] = `
<div>
  
  
  <header
    class="logo"
  >
    
    
    <h1>
      A Bordo
    </h1>
    
    
    <h2>
      A Rede Social dos Velejadores
    </h2>
    
  
  </header>
  
  
  <main
    class="container"
    id="loginPage"
  >
    
    
    <section
      class="img-container"
    />
        
    
    <section
      class="login-container"
      id="login"
    >
      
      
      <header
        class="form-options"
      >
        
        
        <span
          class="opt-login"
        >
          Login
        </span>
        
        
        <span
          class="opt-signup linkSignUp"
        >
          Cadastro
        </span>
        
      
      </header>
      
      
      <section
        class="form-container"
      >
        
        
        <form>
          
          
          <div
            class="form-fields"
          >
            
            
            <label
              for="email"
            >
              E-mail
            </label>
            
            
            <input
              class="input-email form-item"
              id="email"
              type="e-mail"
            />
            
            
            
            <label
              class="label-login"
              for="password"
            >
              Senha
            </label>
            
            
            <input
              class="input-password form-item"
              id="password"
              type="password"
            />
            
          
          </div>
          
          
          <section
            class="errors"
          />
          
          
          <button
            class="btn-signup btn form-item"
            id="signUpButton"
            type="button"
          >
            Cadastrar
          </button>
          
          
          <p
            class="separator"
          >
            ou
          </p>
           
          
          <button
            class="btnGmail btn form-item"
            id="btnGmail"
            type="button"
          >
            
            
            <img
              class="google-icon"
              src="./img/logo-google.png"
            />
            
            
            <span>
              Entrar com o Google
            </span>
            
          
          </button>
          
        
        </form>
        
      
      </section>
      
    
    </section>
    
  
  </main>
  
  
</div>
`;

exports[`reset should load reset page 1`] = `
<div>
  
  
  <header
    class="logo"
  >
    
    
    <h1>
      A Bordo
    </h1>
    
    
    <h2>
      A Rede Social dos Velejadores
    </h2>
    
  
  </header>
  
  
  <main
    class="container"
  >
      
    
    <section
      class="img-container"
    >
      
    
    </section>
        
    
    <section
      class="login-container"
      id="login"
    >
      
      
      <header
        class="reset-text"
      >
        
        Redefinir Senha
      
      </header>
      
      
      <section
        class="form-container"
      >
        
        
        <form>
          
          
          <div
            class="form-fields"
          >
            
            
            <label
              for="email"
            >
              E-mail
            </label>
            
            
            <input
              class="input-item"
              id="email"
              type="e-mail"
            />
            
              
            
            <button
              class="btn-login btn form-item"
              id="buttonReset"
              type="button"
            >
              Enviar
            </button>
            

            
            <button
              class="btn-login btn form-item"
              id="buttonReturn"
              type="button"
            >
              Voltar
            </button>
            
          
          </div>
            
        
        </form>
        
      
      </section>
      
    
    </section>
    
  
  </main>
  
  
</div>
`;
